sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/Hotspots/DoNotHardcodeCredentialsBase.cs

```diff
@@ -54,7 +54,7 @@ public string CredentialWords
                     .Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries)
                     .Select(x => x.Trim())
                     .ToList();
-                this.passwordValuePattern = new Regex(string.Format(@"\b(?<password>{0})\b[:=]\S",
+                this.passwordValuePattern = new Regex(string.Format(@"\b(?<password>{0})\b[:=](?!\?|:|\{{[0-9]*\}})\S",
```

Comment:
The new regex will be more difficult to build. Currently there are some FN:

- Passwords cannot START with `?`, but text like `?whaaat?` is SQL hardcoded password.
- Passwords cannot START with `:`, but text like  `:something:` is SQL hardcoded password. I'm not sure if `:password` and `:param` are the only two compliant examples, or if it's `:\w+`. @pierre-loup-tristant-sonarsource ?
- The `{0}` format string does not consider explicit format `{0:yyMMdd}`. I would suggest `\{\d+[^}]*\}`

Reply:
Just checked with @pierre-loup-tristant-sonarsource and `:param` can actually be any word.